src/Datadog.Trace/RuntimeMetrics/PerformanceCountersListener.cs

```diff
@@ -120,17 +111,33 @@ public void Refresh()
 
         protected virtual void InitializePerformanceCounters()
         {
-            _memoryCategory = new PerformanceCounterCategory(MemoryCategoryName);
+            try
+            {
+                _memoryCategory = new PerformanceCounterCategory(MemoryCategoryName);
 
-            var instanceName = GetInstanceName();
-            _fullInstanceName = instanceName.Item2;
-            _instanceName = instanceName.Item1;
+                var instanceName = GetInstanceName();
+                _fullInstanceName = instanceName.Item2;
+                _instanceName = instanceName.Item1;
+
+                _gen0Size = new PerformanceCounterWrapper(MemoryCategoryName, "Gen 0 heap size", _instanceName);
+                _gen1Size = new PerformanceCounterWrapper(MemoryCategoryName, "Gen 1 heap size", _instanceName);
+                _gen2Size = new PerformanceCounterWrapper(MemoryCategoryName, "Gen 2 heap size", _instanceName);
+                _lohSize = new PerformanceCounterWrapper(MemoryCategoryName, "Large Object Heap size", _instanceName);
+                _contentionCount = new PerformanceCounterWrapper(ThreadingCategoryName, "Total # of Contentions", _instanceName);
+            }
+            catch (UnauthorizedAccessException ex) when (ex.Message.Contains("'Global'"))
+            {
+                // Catching error UnauthorizedAccessException: Access to the registry key 'Global' is denied.
+                // The 'Global' part seems consistent across localizations
 
-            _gen0Size = new PerformanceCounterWrapper(MemoryCategoryName, "Gen 0 heap size", _instanceName);
-            _gen1Size = new PerformanceCounterWrapper(MemoryCategoryName, "Gen 1 heap size", _instanceName);
-            _gen2Size = new PerformanceCounterWrapper(MemoryCategoryName, "Gen 2 heap size", _instanceName);
-            _lohSize = new PerformanceCounterWrapper(MemoryCategoryName, "Large Object Heap size", _instanceName);
-            _contentionCount = new PerformanceCounterWrapper(ThreadingCategoryName, "Total # of Contentions", _instanceName);
+                Log.Error(ex, "The process does not have sufficient permissions to read performance counters. Please refer to https://docs.datadoghq.com/tracing/runtime_metrics/dotnet/#additional-permissions-for-iis to learn how to grant those permissions.");
```

Comment:
Do you know if we can create public short links? It'd be great to have one for this purpose, in case the web page changes in any way

Reply:
That would be great, but I'm not aware of a datadog url shortener 🤔